File: tests/machine/x/zig/nested_function.zig
const std = @import("std");

fn outer(x: i32) i32 {
    const inner = (struct { x: i32, fn call(self: @This(), y: i32) i32 {
        return (self.x + y);
} }{ .x = x }).call;
    return inner(5);
}

pub fn main() void {
    std.debug.print("{d}\n", .{outer(3)});
}

File: tests/machine/x/pascal/nested_function.pas
program main;
{$mode objfpc}

uses SysUtils, fgl, fphttpclient, Classes, Variants, fpjson, jsonparser;

type
  generic TArray<T> = array of T;

function outer(x: integer): integer;
begin
  result := inner(5);
  exit;
end;

begin
  writeln(outer(3));
end.
